model-optimizer/extensions/front/onnx/dequantize_linear_resolver.py

```diff
@@ -0,0 +1,73 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import numpy as np
+
+from extensions.front.onnx.quantize_dequantize_linear import QuantizeDequantizeLinear
+from extensions.ops.Cast import Cast
+from extensions.ops.elementwise import Mul, Sub
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementOp
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.ops.broadcast import Broadcast
+from mo.ops.shape import Shape
+
+
+class DequantizeLinearResolver(FrontReplacementOp):
+    """
+    DequantizeLinear can be replace with the following formula: y = (x - x_zero_point) * x_scale
+    """
+    op = "DequantizeLinear"
+    enabled = True
+
+    def run_after(self):
+        return [QuantizeDequantizeLinear]
+
+    def replace_op(self, graph: Graph, node: Node):
+        node_name = node.soft_get('name', node.id)
+        cast = Cast(graph, {'dst_type': np.float32, 'name': node_name + '/Cast'}).create_node()
+        node.in_port(0).get_connection().set_destination(cast.in_port(0))
+        mul = Mul(graph, {}).create_node()
+
+        if node.has_valid('axis'):
```

Comment:
Please, provide some comments describing what this transformation does and why.

nit: Also we have several checks for `node.has_valid('axis')`. Can we restructure the code to make it simpler?

Reply:
I think comments will make it more understandable, not sure how to make it more simple